### ISSUE TITLE
improve optimizer

### ISSUE THREAD

**Issue by megbedell** [megbedell/wobble#8]
improve gradient descent & add some convergence criterion to the optimize_order() function.

**Comment by davidwhogg** [comment]
We want to do some adaptive step-size stuff. I can help with that.

**Comment by megbedell** [comment]
all hail tensorflow
